examples/weak.rs

```diff
@@ -0,0 +1,37 @@
+extern crate froggy;
+use froggy::{Storage, WeakPointer};
+
+struct Node {
+    value: String,
+    next: Option<WeakPointer<Node>>,
+}
+
+impl Drop for Node {
+    fn drop(&mut self) {
+        println!("{} destroyed", &self.value);
+    }
+}
+
+fn main() {
+    let storage = Storage::new();
+    let node1 = storage.write().create(Node {
+        value: "Node 1".to_string(),
+        next: None,
+    });
+    let node2 = storage.write().create(Node {
+        value: "Node 2".to_string(),
+        next: None,
+    });
+
+    storage.write().access_mut(&node1).next = Some(node2.downgrade());
+    storage.write().access_mut(&node2).next = Some(node1.downgrade());
+
+    for node in storage.read().iter() {
+        let value = node.next.as_ref().map_or("None".into(), |ref next| {
+            next.upgrade().ok().map_or("None".into(), |ref ptr| {
```

Comment:
can't we just `unwrap` here, considering that we know the storage and target elements are alive?